### PR TITLE
add pytest tests and tests for datatables api

### PR DESCRIPTION
**Pull request by lime-green**
- Adds the pytest library (https://docs.pytest.org/en/latest/contents.html) which is a modern library that is very nice to use
- Adds tests for the datatables API as a starting point
- now only tests on python 3.7 (previously 3.5 + 3.6) to match prod

usage:

```
# to install the additional dependencies:
pip install -r requirements_dev.txt

# to run the test suite
pytest

# if you need to update snapshots
pytest --snapshot-update

# if you need a new DB
pytest --create-db
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,18 +4,18 @@ dist: bionic
 sudo: false
 
 python:
-  - '3.5'
+  # TODO: Make sure 3.6 is what is running on the servers
```
Comment (lime-green) [inline .travis.yml:7]: ^ needs confirmation

Comment (lime-green) [reply .travis.yml:7]: @raylu confirmed 3.7

File: .travis.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ dist: bionic
...
```
Comment (climu) [inline .travis.yml]:
```
(venv) climu@openstudyroom:~/osr/venv/bin$ python --version
Python 3.5.3
```

File: .travis.yml (hunk 1 above)
Comment (climu) [reply .travis.yml]: Really?

Comment (raylu) [reply .travis.yml]: as mentioned amidst all my discord spam, we're 3.7 everywhere on DO, even for osr-bot. py3.5 has been uninstalled

**Pull request merged**